src/core/engine/processors/mongo_to_postgresql_order_status_history.py

```diff
@@ -0,0 +1,92 @@
+from src.core.engine.processors.abstract_etl_processor import AbstractEtl
+from src.lib.entities.mongo_engine_odm_mapping import (
+    OrderStatusHistory as MongoOrderStatusHistory,
+)
+from src.lib.entities.sqlalchemy_orm_mapping import (
+    OrderStatusHistory as SqlalchemyOrderStatusHistory,
+)
+from src.utils.etl_util import pass_values_of_odm_object_to_orm_object
+
+
+class MongoToPostgresqlOrderStatusHistory(AbstractEtl):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+
+        self.mongo_connection = None
+        self.postgresql_session = None
+        self.last_order_status_history_document_created_date = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_connection = ioc.get_instance("mongo_engine_connection")
+        self.postgresql_session = ioc.get_instance("sqlalchemy_session")
+
+    def extract_data(self):
+        extracted_data = {}
+
+        if MongoOrderStatusHistory.objects.count() == 0:
+            print("empty collection")
+
+        if not self.last_order_status_history_document_created_date:
+            first_order_status_history = (
+                MongoOrderStatusHistory.objects().order_by("from_time").limit(1)
+            )
+            self.last_order_status_history_document_created_date = (
+                first_order_status_history[0].created_date
+            )
+            extracted_data[
+                first_order_status_history[0].id
+            ] = first_order_status_history[0]
+
+        last_order_status_history = (
+            MongoOrderStatusHistory.objects().order_by("-from_time").limit(1)
+        )
+
+        if (
+            self.last_order_status_history_document_created_date
+            == last_order_status_history[0].created_date
+        ):
+            return
+
+        order_status_histories_from_mongo = MongoOrderStatusHistory.objects(
+            from_time__gt=self.last_order_status_history_document_created_date
+        ).order_by("created_date")
+
+        self.last_order_status_history_document_created_date = (
+            order_status_histories_from_mongo[
+                order_status_histories_from_mongo.count() - 1
+            ].created_date
+        )
+
+        for order_status_history in order_status_histories_from_mongo:
```

Comment:
You don't need to return a map, since you will iterate through all the records. A list or a set are better data structures.

Reply:
done.

Reply:
done